coriolis/tasks/osmorphing_tasks.py

```diff
@@ -14,6 +16,55 @@
 LOG = logging.getLogger(__name__)
 
 
+def _reorder_root_disk(volumes_info, root_device, os_type):
+    """
+    Reorders volumes_info so that the root disk will always be the first volume
+
+    root_device is returned by the OSMount Tools as the root partition device
+    (i.e. /dev/vdd2 for linux).
+
+    For Linux, we need to strip the trailing digits
+    to get the actual disk device. After that, we convert the last letter of
+    the disk device name into the equivalent index by alphabetical order.
+
+    Windows OSMount Tools should directly return the root disk index extracted
+    from diskpart (as string; i.e. '1').
+
+    Reordering is done by swapping the indexes of the volumes_info list.
+    """
+    if not root_device:
+        LOG.warn('os_root_dev was not returned by OSMount Tools. '
+                 'Skipping root disk reordering')
+        return
+
+    # the default disk device of the migrated VM should be /dev/Xdb, because
+    # /dev/Xda should be the worker's root disk. Same for Windows: Disk 0
+    # should be the worker's root disk, and Disk 1 the migrated VM's root disk
+    linux_root_default = 'b'
+    windows_root_default = 1
+
+    supported_os_types = [constants.OS_TYPE_LINUX, constants.OS_TYPE_WINDOWS]
+    if os_type == constants.OS_TYPE_LINUX:
+        pattern = r'[0-9]'
+        root_disk = re.sub(pattern, '', root_device)
+        disk_index = ord(root_disk[-1]) - ord(linux_root_default)
```

Comment:
This may seem like a stretch but this code will completely miss its mark in case of clouds where separate device models would dictate a separate device name prefix (e.g. a setup where 2-3 disks are "/dev/vdN" and 2-3 are "/dev/sdn".

From my PoV there is no sane way of handling this without having the providers knowing and passing the disk device mapping (i.e. "Disk ID %s is /dev/sdc") from the DEPLOY_OS_MORPHING_RESOURCES task directly.